File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> hashMap;
        vector<vector<string>> ans;
        for(auto str: strs){
            string temp = str;
            sort(temp.begin(), temp.end());
            hashMap[temp].push_back(str);
        }
        for(auto it: hashMap){
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();

        for(String str: strs){
            int[] keyArray = new int[26];
            for(Character ch: str.toCharArray()){
                keyArray[ch - 'a'] += 1;
            }
            String key = Arrays.toString(keyArray);
            map.computeIfAbsent(key, k -> new ArrayList<>()).add(str);
        }

        return map.values().stream().collect(Collectors.toList());
    }
}